### PR TITLE
Add dual-stack support to HostedCluster

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
Enables support for IPv6 in HostedCluster API. Deprecates existing
scalar values and replaces them with slices that mirror network
settings in InstallConfig.

Replaces individual fields for networking in the HostedControlPlane spec
with the networking struct of the HostedCluster spec.

These changes enable parity at the API level with standalone OpenShift

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-500](https://issues.redhat.com//browse/HOSTEDCP-500)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [x] This change includes unit tests.

### CODE REVIEW

**Comment by csrwng** [comment]
Supersedes https://github.com/openshift/hypershift/pull/1177

File: cmd/cluster/cluster.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ func NewCreateCommands() *cobra.Command {
 	cmd.PersistentFlags().StringVar(&opts.InfrastructureJSON, "infra-json", opts.InfrastructureJSON, "Path to file containing infrastructure information for the cluster. If not specified, infrastructure will be created")
 	cmd.PersistentFlags().StringVar(&opts.InfraID, "infra-id", opts.InfraID, "Infrastructure ID to use for hosted cluster resources.")
 	cmd.PersistentFlags().StringVar(&opts.ServiceCIDR, "service-cidr", opts.ServiceCIDR, "The CIDR of the service network")
-	cmd.PersistentFlags().StringVar(&opts.PodCIDR, "pod-cidr", opts.PodCIDR, "The CIDR of the pod network")
+	cmd.PersistentFlags().StringVar(&opts.ClusterCIDR, "pod-cidr", opts.ClusterCIDR, "The CIDR of the pod network")
```
Comment (enxebre) [inline cmd/cluster/cluster.go:64]: can we align the flag name with the API field name? either use pod* or cluster* on both?

File: cmd/infra/aws/create.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ type CreateInfraOutput struct {
 	Region          string                   `json:"region"`
 	Zone            string                   `json:"zone"`
 	InfraID         string                   `json:"infraID"`
-	ComputeCIDR     string                   `json:"computeCIDR"`
+	MachineCIDR     string                   `json:"computeCIDR"`
```
Comment (enxebre) [inline cmd/infra/aws/create.go:52]: replace `json:"computeCIDR"` / machineCIDR

File: control-plane-operator/controllers/hostedcontrolplane/kas/params.go [hunk 3]
```diff
@@ -40,8 +40,8 @@ type KubeAPIServerParams struct {
 	CloudProviderCreds  *corev1.LocalObjectReference `json:"cloudProviderCreds"`
 
 	ServiceAccountIssuer string                       `json:"serviceAccountIssuer"`
-	ServiceCIDR          string                       `json:"serviceCIDR"`
-	PodCIDR              string                       `json:"podCIDR"`
+	ServiceCIDRs         []string                     `json:"serviceCIDR"`
+	ClusterCIDRs         []string                     `json:"podCIDR"`
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/kas/params.go:44]: `json:"serviceCIDR"` -> serviceCIDRs
`json:"podCIDR"` -> `json:"clusterCIDRs"`

**Comment by enxebre** [comment]
Thanks @csrwng drop some small comments, looks great!
PTAL @danwinship or please tag some one from networking team.

**Comment by csrwng** [comment]
Thanks, squashed

**Comment by csrwng** [comment]
@enxebre this is squashed (still 2 commits to keep the dependency bump separate), and is passing tests, ptal

File: api/v1alpha1/hosted_controlplane.go [hunk 4]
```diff
@@ -34,30 +34,37 @@ type HostedControlPlaneSpec struct {
...
-	MachineCIDR string `json:"machineCIDR"`
+	// +optional
+	MachineCIDR string `json:"machineCIDR,omitempty"`
 
 	// deprecated
 	// use networking.NetworkType
 	// NetworkType specifies the SDN provider used for cluster networking.
-	NetworkType NetworkType                 `json:"networkType"`
-	SSHKey      corev1.LocalObjectReference `json:"sshKey"`
+	// +optional
```
Comment (enxebre) [inline api/v1alpha1/hosted_controlplane.go:53]: why is this optional now?

**Pull request merged**